core/src/main/java/org/elasticsearch/common/geo/ShapeRelation.java

```diff
@@ -37,6 +42,37 @@
         this.relationName = relationName;
     }
 
+    @Override
+    public ShapeRelation readFrom(StreamInput in) throws IOException {
```

Comment:
for my future reference (we had a similar discussion just today on some other PR), why don't we simply rely on the enum `ordinal()` here like we already do elsewhere?